CONTRIBUTING.md

```diff
@@ -0,0 +1,884 @@
+the [Common prerequisites and dependencies](), including installing
+`pytest`, as you will need it for the sections described below.
+
+#### Writing Python Tests
+
+If your PR includes code that isn't currently covered by our tests (e.g.
+adding a new flavor, adding autolog support to a flavor, etc.), you
+should write tests that cover your new code. Your tests should be added
+to the relevant file under `tests`, or if there is no appropriate file,
+in a new file prefixed with `test_` so that `pytest` includes that file
+for testing.
+
+If your tests require usage of a tracking URI, the [pytest
+fixture](https://docs.pytest.org/en/3.2.1/fixture.html)
+[tracking\_uri\_mock](https://github.com/mlflow/mlflow/blob/master/tests/conftest.py#L74)
+is automatically set up for every tests. It sets up a mock tracking URI
+that will set itself up before your test runs and tear itself down
+after.
+
+By default, runs are logged under a local temporary directory that's
+unique to each test and torn down immediately after test execution. To
+disable this behavior, decorate your test function with
+`@pytest.mark.notrackingurimock`
+
+#### Running Python Tests
+
+Verify that the unit tests & linter pass before submitting a pull
+request by running:
+
+We use [Black](https://black.readthedocs.io/en/stable/) to ensure a
+consistent code format. You can auto-format your code by running:
+
+``` bash
+black .
+```
+
+Then, verify that the unit tests & linter pass before submitting a pull
+request by running:
+
+``` bash
+./dev/lint.sh
+./dev/run-python-tests.sh
+```
+
+We use [pytest](https://docs.pytest.org/en/latest/contents.html) to run
+Python tests. You can run tests for one or more test directories or
+files via `pytest [file_or_dir] ... [file_or_dir]`. For example, to run
+all pyfunc tests, you can run:
+
+``` bash
+pytest tests/pyfunc
+```
+
+Note: Certain model tests are not well-isolated (can result in OOMs when
+run in the same Python process), so simply invoking `pytest` or `pytest
+tests` may not work. If you'd like to run multiple model tests, we
+recommend doing so via separate `pytest` invocations, e.g. `pytest
+tests/sklearn && pytest tests/tensorflow`
+
+If opening a PR that changes or adds new APIs, please update or add
+Python documentation as described in [Writing Docs](#writing-docs) and
+commit the docs to your PR branch.
+
+#### Python Client
+
+For the client, if you are adding new model flavors, follow the
+instructions below.
+
+##### Python Model Flavors
+
+If you are adding new framework flavor support, you'll need to modify
+`pytest` and Github action configurations so tests for your code can run
+properly. Generally, the files you'll have to edit are:
+
+1.  `dev/run-python-tests.sh`:
+
+> 1.  Add your tests to the ignore list, where the other frameworks are
+>     ignored
+> 2.  Add a pytest command for your tests along with the other framework
+>     tests (as a separate command to avoid OOM issues)
```

Comment:
```suggestion
    1. Add your tests to the ignore list, where the other frameworks are
       ignored
    2. Add a pytest command for your tests along with the other framework
       tests (as a separate command to avoid OOM issues)
```